rbd.c

```diff
@@ -239,6 +244,54 @@ static void tcmu_rbd_service_status_update(struct tcmu_device *dev,
 
 #endif /* LIBRADOS_SUPPORTS_SERVICES */
 
+static int tcmu_rm_stale_entries_from_blacklist(struct tcmu_device *dev)
+{
+	struct tcmu_rbd_state *state = tcmur_dev_get_private(dev);
+	char *cmd = NULL;
+	char **entry, *addr;
+	int ret = 0;
+	int i = 0, j;
+
+	pthread_mutex_lock(&blacklist_caches_lock);
+	if (darray_empty(blacklist_caches))
+		goto unlock;
+
+	/* Try to remove all the stale blacklist entities */
+	darray_foreach(entry, blacklist_caches) {
+		tcmu_dev_info(dev, "removing addrs: {%s}\n", *entry);
+		ret = asprintf(&cmd,
+			      "{\"prefix\": \"osd blacklist\","
+			      "\"blacklistop\": \"rm\","
+			      "\"addr\": \"%s\"}", *entry);
+		if (ret < 0) {
+			tcmu_dev_warn(dev, "Could not allocate command. (Err %d)\n",
+				      ret);
+			free(cmd);
+			goto unlock;
+		}
+		ret = rados_mon_command(state->cluster, (const char**)&cmd, 1, NULL, 0, NULL, NULL,
+					NULL, NULL);
+		if (ret < 0) {
+			tcmu_dev_err(dev, "Could not rm blacklist entry '%s'. (Err %d)\n",
+				     *entry, ret);
+			free(cmd);
+			goto unlock;
```

Comment:
Nit: should it still keep trying the other addresses?

Reply:
Do you mean the `cmd`, right ? 

It will be alloced from asprint() with the size needed every time. BTW, will the `nonce` 
 have different lengths for different sessions some times ? Or will they always have a same string length ? 

Maybe we could just alloc a memory and to reuse it.

Reply:
I just mean that it you stop iterating through the `darray` once you hit the first error.

Reply:
I think I have lost my last comments about this , and I had fixed it already. Thanks.